src/ucs/datastruct/callbackq.c

```diff
@@ -247,3 +248,73 @@ ucs_status_t ucs_callbackq_remove_safe(ucs_callbackq_t *cbq, ucs_callback_t cb,
     ucs_callbackq_leave(cbq);
     return UCS_OK;
 }
+
+void ucs_callbackq_slow_elem_init(ucs_callbackq_slow_elem_t *elem)
+{
+#if ENABLE_ASSERT
+    elem->is_added = 0;
+#endif
+}
+
+static void ucs_callbackq_slow_path_cb(void *arg)
+{
+    ucs_callbackq_t *cbq = arg;
+    ucs_callbackq_slow_elem_t *elem, *tmp_elem;
+
+    ucs_callbackq_enter(cbq);
+
+    /* TODO avoid locks on every iteration. For instance implement
+     * lazy remove with additional list of elements to be removed. */
+    ucs_list_for_each_safe(elem, tmp_elem, &cbq->slow_path, list) {
+        ucs_callbackq_leave(cbq);
+        elem->cb(elem);
+        ucs_callbackq_enter(cbq);
+    }
+
+    ucs_callbackq_leave(cbq);
+}
+
+/* Note: multiple calls with the same element are not allowed */
+void ucs_callbackq_add_slow_path(ucs_callbackq_t *cbq,
+                                 ucs_callbackq_slow_elem_t* elem)
+{
+    ucs_status_t status;
+
+    ucs_callbackq_enter(cbq);
+
+    /* Coverity complains on  ucs_assert(++elem_is_added !=1),
+     * so use clear macro */
+#if ENABLE_ASSERT
+    if (++elem->is_added != 1) {
```

Comment:
how about
ucs_assertv(++elem->is_added == 0, "Slow path element can be added to callbackq just once")